src/event-creator.ts

```diff
@@ -0,0 +1,34 @@
+import { SDKEvent } from './sdkRuntimeModels';
+import { IMParticleWebSDKInstance } from './mp-instance';
+import Constants from './constants';
+import { EventType } from './types';
+
+export function createBaseEvent(
+    mpInstance: IMParticleWebSDKInstance,
+    messageType: number,
+    eventName: string = ''
+): SDKEvent {
+    const now = Date.now();
+    const { _Store, Identity, _timeOnSiteTimer } = mpInstance;
+    const { sessionId, deviceId, sessionStartDate, SDKConfig } = _Store;
+    return {
+        EventDataType: messageType,
+        EventName: eventName,
+        EventCategory: EventType.Other,
+        Timestamp: now,
+        SessionId: sessionId,
+        MPID: Identity.getCurrentUser()?.getMPID(),
+        DeviceId: deviceId,
+        IsFirstRun: false,
+        SourceMessageId: '',
```

Comment:
Same here. Use `utils.generateUniqueId`

Reply:
this file actually shouldn't be here.  removing.